chapter08/jxp/ch8ex3.pl

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use File::Spec::Functions qw(canonpath);
+
+sub printdir {
+  my ( $dirname ) = @_;
+  opendir my $dir_fh, $dirname or die "Cannot open directory";
+  foreach my $file ( readdir( $dir_fh ) ) {
+     print canonpath($dirname),"/",$file, "\n";
+  }
+}
```

Comment:
The question wanted us to pass in the directory handle as an argument to the `printdir` subroutine, rather than just the name.